src/db_adapters/blocks.rs

```diff
@@ -16,30 +15,15 @@ pub(crate) async fn store_block(
 ) {
     let block_model = models::blocks::Block::from(block);
 
-    let mut interval = crate::INTERVAL;
-    loop {
-        match diesel::insert_into(schema::blocks::table)
+    crate::execute_retriable_or_panic!(
+        diesel::insert_into(schema::blocks::table)
             .values(block_model.clone())
             .on_conflict_do_nothing()
-            .execute_async(&pool)
-            .await
-        {
-            Ok(_) => break,
-            Err(async_error) => {
-                error!(
-                    target: crate::INDEXER_FOR_EXPLORER,
-                    "Error occurred while Block was adding to database. Retrying in {} milliseconds... \n {:#?} \n {:#?}",
-                    interval.as_millis(),
-                    async_error,
-                    &block_model
-                );
-                tokio::time::sleep(interval).await;
-                if interval < crate::MAX_DELAY_TIME {
-                    interval *= 2;
-                }
-            }
-        }
-    }
+            .execute_async(&pool),
+        100,
```

Comment:
Adjust this number so we don't stuck for more than 5 minutes. By that time we should be notified about the failure.